content/articles/creation-of-calendar-heatmaps-to-display-time-series-quantitative-data-with-python/index.md

```diff
@@ -0,0 +1,115 @@
+```py
+#changing the format of the data type   
+df['won_date'] = pd.to_datetime(df['won_date'])  
+#Setting won date as index  
+df.set_index('won_date', inplace = True)
+```
+###  Setting up the Calendar Plot with the Library Calplot Modules
+The calplot function is used for the generation of this plot. To create the plot the first argument of the variable and data frame is defined by the function calplot. **how** shows how we want to wholly form the data since it is the second argument. More standard python functions can still be passed.
+The following argument to be passed is the **cmap** that defines the color scheme to be applied which is the color map.
+```py
+import calplotpl1 = calplot.calplot(data = df['declared_monthly_revenue'],how = 'sum', cmap = 'Reds', figsize
+= (18, 9), suptitle = "Total declared monthly revenue")
+```
+Time series is thus divided into into years and the to months. We see that on  more dark colours their is more revenue collected  as opposed more light colours. Making it easier to identify most sales in a month.
+Let us give a try to a new calplot library counting number of  won date put on a daily basis. We will also understand it more by the code snippet below.
+```py
+#We will be grouping won orders by their ID and have them counted daily.
+counts = df.groupby('won_date')['declared_monthly_revenue'].agg( 'count').reset_index()  
+counts['won_date'] = pd.to_datetime(counts['won_date'])  
+counts# How creation of the plot is done.
+calplot.calplot(counts['declared_monthly_revenue'], cmap = 'GnBu', textformat  ={:.0f}', figsize = (18, 9), suptitle = "Total declared monthly revenue")
+```
+Our results can be displayed as illustrated in the image bellow:
+![daily](/engineering-education/how-to-perform-threading-timer-in-python/daily.png)
+
+Since our dataset only covers  one year, it only executes a year's time series instead of multiple series of years.
+
+###  Calendar Heatmap using the Python library ‘july’
+This function displays one year per month of data rather than many years.
+For us to use this library, we first have it installed and the dates to be output calculated in range. Important functions to consider are **heatmap** which creates the consolidated plot annually while the **calendar_plot**   displays each month  separately from the plotted year. More illustrations by code are implemented below:
+ You can have module july first installed
+ ```py
+  pip install july
+ ```
+ After succesfully having it installed you can now procede to the next step:
+ ```py
+ import july  
+from july.utils import date_range  
+dates = date_range("2018-01-01", "2018-12-31")  
+july.heatmap( dates, data =df1['declared_monthly_revenue'], title='Total declared monthly revenue', cmap="golden", month_grid=True, horizontal = True)
+```
+
+We will now be using calendar_plot as a function as a subset of july.
+```py
+july.calendar_plot(dates, df1['declared_monthly_revenue'], cmap = 'copper');
+```
```

Comment:
![copy7](https://user-images.githubusercontent.com/50293753/147403118-a8885c4a-1569-4222-89de-e0431583c131.png)

Reply:
Hello @GeekEdmund 
I find your article not unique, it is just like a replica of this article:
[Calendar Heatmaps : A perfect way to display your time-series quantitative data](https://medium.com/analytics-vidhya/calendar-heatmaps-a-perfect-way-to-display-your-time-series-quantitative-data-ad36bf81a3ed)

@ahmadmardeni1 @hectorkambow 
I need your help on this article, the author says he used the other article as a source of reference.